packages/graphql/src/translate/queryAST/ast/QueryAST.ts

```diff
@@ -18,28 +18,40 @@
  */
 
 import Cypher from "@neo4j/cypher-builder";
-import type { ReadOperation } from "./operations/ReadOperation";
+import { ReadOperation } from "./operations/ReadOperation";
 import type { QueryASTNode } from "./QueryASTNode";
 import { QueryASTContext, QueryASTEnv } from "./QueryASTContext";
 import { createNodeFromEntity } from "../utils/create-node-from-entity";
 import type { Neo4jGraphQLContext } from "../../../types/neo4j-graphql-context";
+import type { CompositeReadOperation } from "./operations/composite/CompositeReadOperation";
 
 export class QueryAST {
-    private operation: ReadOperation;
+    private operation: ReadOperation | CompositeReadOperation;
```

Comment:
Maybe it's just worth changing to
```suggestion
    private operation: Operation;
```